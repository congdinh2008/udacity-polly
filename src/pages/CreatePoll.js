const CreatePoll = () => {
  return <h1>CreatePoll</h1>;
};

export default CreatePoll;
