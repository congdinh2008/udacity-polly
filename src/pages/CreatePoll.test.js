/* eslint-disable */
import { fireEvent, render, screen } from "@testing-library/react";
import React from "react";
import CreatePoll from "./CreatePoll";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import { legacy_createStore as createStore } from "redux";
import reducer from "../reducers";
import middleware from "../middleware";

describe("CreatePoll", () => {
  const store = createStore(reducer, middleware);

  it("will match snapshot", () => {
    const view = render(
      <Provider store={store}>
        <Router>
          <CreatePoll />
        </Router>
      </Provider>
    );

    expect(view).toMatchSnapshot();
  });

  it("will show 2 inputs on page", () => {
    render(
      <Provider store={store}>
        <Router>
          <CreatePoll />
        </Router>
      </Provider>
    );
    const optionOneInput = screen.getByPlaceholderText("Option One");
    const optionTwoInput = screen.getByPlaceholderText("Option Two");
    const createButton = screen.getByText("Create Poll");

    expect(optionOneInput).toBeInTheDocument();
    expect(optionTwoInput).toBeInTheDocument();
    expect(createButton).toBeInTheDocument();
  });

  it("disable create button if all inputs are empty", () => {
    render(
      <Provider store={store}>
        <Router>
          <CreatePoll />
        </Router>
      </Provider>
    );
    const createButton = screen.getByText("Create Poll");
    expect(createButton).toBeDisabled();
  });

  it("enable create button if all inputs are not empty", () => {
    render(
      <Provider store={store}>
        <Router>
          <CreatePoll />
        </Router>
      </Provider>
    );

    const optionOneInput = screen.getByPlaceholderText("Option One");
    const optionTwoInput = screen.getByPlaceholderText("Option Two");
    fireEvent.change(optionOneInput, {
      target: {
        value: "Learning English",
      },
    });
    fireEvent.change(optionTwoInput, {
      target: {
        value: "Learning Mathematics",
      },
    });
    const submitQuestionButton = screen.getByText("Create Poll");
    expect(submitQuestionButton).not.toBeDisabled();
  });
});
