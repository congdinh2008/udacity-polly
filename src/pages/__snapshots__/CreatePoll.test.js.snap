// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePoll will match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="form-section d-flex flex-colum justify-content-center"
      >
        <div
          class="form-container create-poll-container text-center"
        >
          <div
            class="form-header"
          >
            <h1>
              Would You Rather
            </h1>
            <span>
              Create Your Own Poll
            </span>
          </div>
          <div
            class="form create-poll-form"
          >
            <form>
              <div
                class="form-group d-flex flex-column"
              >
                <label
                  class="form-control-label"
                  for="option-one"
                >
                  First Option
                </label>
                <input
                  id="option-one"
                  placeholder="Option One"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="form-group d-flex flex-column"
              >
                <label
                  class="form-control-label"
                  for="option-two"
                >
                  Second Option
                </label>
                <input
                  id="option-two"
                  placeholder="Option Two"
                  type="text"
                  value=""
                />
              </div>
              <button
                class="btn btn-submit"
                disabled=""
                type="submit"
              >
                Create Poll
              </button>
            </form>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="form-section d-flex flex-colum justify-content-center"
    >
      <div
        class="form-container create-poll-container text-center"
      >
        <div
          class="form-header"
        >
          <h1>
            Would You Rather
          </h1>
          <span>
            Create Your Own Poll
          </span>
        </div>
        <div
          class="form create-poll-form"
        >
          <form>
            <div
              class="form-group d-flex flex-column"
            >
              <label
                class="form-control-label"
                for="option-one"
              >
                First Option
              </label>
              <input
                id="option-one"
                placeholder="Option One"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group d-flex flex-column"
            >
              <label
                class="form-control-label"
                for="option-two"
              >
                Second Option
              </label>
              <input
                id="option-two"
                placeholder="Option Two"
                type="text"
                value=""
              />
            </div>
            <button
              class="btn btn-submit"
              disabled=""
              type="submit"
            >
              Create Poll
            </button>
          </form>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
